include/modules/httpd.h

```diff
@@ -25,11 +25,51 @@
 
 #include "base.h"
 #include "event.h"
+#include "http_parser.h"
 
 #include <string>
 #include <sstream>
 #include <map>
 
+struct HTTPRequestURI
+{
+	std::string path;
+	insp::flat_multimap<std::string, std::string> query_params;
+	std::string fragment;
+
+	HTTPRequestURI(const std::string& uri)
+	{
+		http_parser_url url;
+		http_parser_url_init(&url);
+		http_parser_parse_url(uri.c_str(), uri.size(), 0, &url);
```

Comment:
Can all of this code be removed from the public header? I'd like to avoid having internal dependencies leak into the ABI.